sagemaker-python-sdk/mxnet_onnx_segmentation/mxnet-onnx-duc.ipynb

```diff
@@ -0,0 +1,464 @@
+    "\n",
+    "You can read more about creating an `MXNetModel` object in the [SageMaker Python SDK API docs](https://sagemaker.readthedocs.io/en/stable/sagemaker.mxnet.html#mxnet-model)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.mxnet import MXNetModel\n",
+    "\n",
+    "mxnet_model = MXNetModel(model_data=model_data,\n",
+    "                         entry_point='resnet-duc-hdc.py',\n",
+    "                         role=role,\n",
+    "                         framework_version='1.3.0')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating an inference endpoint\n",
+    "\n",
+    "Now we can use our `MXNetModel` object to build and deploy an `MXNetPredictor`. This creates a SageMaker Model and Endpoint, the latter of which we can use for performing inference. \n",
+    "\n",
+    "We pass the following arguments to the deploy method:\n",
+    "\n",
+    "* `instance_count` - how many instances to back the endpoint.\n",
+    "* `instance_type` - which EC2 instance type to use for the endpoint. For information on supported instance, please check here.\n",
+    "\n",
+    "### How our models are loaded\n",
+    "By default, the predefined SageMaker MXNet containers have a default `model_fn`, which determines how your model is loaded. The default `model_fn` loads an MXNet Module object with a context based on the instance type of the endpoint.\n",
+    "\n",
+    "### Choosing instance types\n",
+    "We will deploy this model with instance type `ml.c5.xlarge`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "predictor = mxnet_model.deploy(initial_instance_count=1, instance_type='ml.c5.xlarge')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Performing inference\n",
+    "\n",
+    "With our Endpoint deployed, we can now send inference requests to it. We'll use one image as an example here."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Preparing the image\n",
+    "\n",
+    "First, we'll download the image (and view it)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "data = 'https://s3.amazonaws.com/onnx-model-zoo/duc/city1.png'\n",
+    "HTML(open(\"input.html\").read())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Download test image\n",
+    "print ('URL: ' + data)\n",
+    "mx.test_utils.download(data)\n",
+    "# read image as rgb\n",
+    "im = cv.imread(os.path.basename(data))[:, :, ::-1]\n",
+    "im = cv.resize(im, (224,224))\n",
```

Comment:
Can we support custom image size with hosting file itself rather than resizing the image? Resizing to smaller shape might cause some unclear output.

Reply:
SageMaker doesn't seem to have custom APIs to specify a different input size on bind for every inference call. Exploring if there is a workaround. Otherwise I will add a comment telling the user to change the bind call's input data shape to match the input image size, for every inference call.